From e9990a3dccdd5681a239ad2a50fc0d6ecd51751e Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:47 -0400
Subject: [fs] nfsd: remove the client_mutex and the nfs4_lock/unlock_state wrappers

Message-id: <1410898849-22126-164-git-send-email-bfields@redhat.com>
Patchwork-id: 95797
O-Subject: [Patch RHEL7 163/165] nfsd: remove the client_mutex and the nfs4_lock/unlock_state wrappers
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: b687f6863eed050aa56fe176e513025907ecd287
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index b900af8..2e80a59 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -75,9 +75,6 @@ static void nfs4_free_ol_stateid(struct nfs4_stid *stid);
 
 /* Locking: */
 
-/* Currently used for almost all code touching nfsv4 state: */
-static DEFINE_MUTEX(client_mutex);
-
 /*
  * Currently used for the del_recall_lru and file hash table.  In an
  * effort to decrease the scope of the client_mutex, this spinlock may
@@ -97,12 +94,6 @@ static struct kmem_cache *file_slab;
 static struct kmem_cache *stateid_slab;
 static struct kmem_cache *deleg_slab;
 
-void
-nfs4_lock_state(void)
-{
- mutex_lock(&client_mutex);
-}
-
 static void free_session(struct nfsd4_session *);
 
 static bool is_session_dead(struct nfsd4_session *ses)
@@ -118,12 +109,6 @@ static __be32 mark_session_dead_locked(struct nfsd4_session *ses, int ref_held_b
  return nfs_ok;
 }
 
-void
-nfs4_unlock_state(void)
-{
- mutex_unlock(&client_mutex);
-}
-
 static bool is_client_expired(struct nfs4_client *clp)
 {
  return clp->cl_time == 0;
diff --git a/fs/nfsd/state.h b/fs/nfsd/state.h
index a02358f..32a7c29 100644
--- a/fs/nfsd/state.h
+++ b/fs/nfsd/state.h
@@ -437,8 +437,6 @@ struct nfsd_net;
 extern __be32 nfs4_preprocess_stateid_op(struct net *net,
   struct nfsd4_compound_state *cstate,
   stateid_t *stateid, int flags, struct file **filp);
-extern void nfs4_lock_state(void);
-extern void nfs4_unlock_state(void);
 void nfs4_put_stid(struct nfs4_stid *s);
 void nfs4_remove_reclaim_record(struct nfs4_client_reclaim *, struct nfsd_net *);
 extern void nfs4_release_reclaim(struct nfsd_net *);
-- 
1.7.1